### PR TITLE
DHFPROD-5850:Pass sequence to custom SJS mapping function

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Pass sequence to custom SJS mapping function
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/function-metadata.xsl [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                <xsl:attribute name="href">
+                    <xsl:value-of select="$location"/>
+                </xsl:attribute>
+                <xsl:for-each select="mapping:parameters/mapping:parameter">
+                    <axsl:with-param axdmp:dialect="1.0-ml">
+                        <xsl:attribute name="name">
+                            <xsl:value-of select="@name"/>
+                        </xsl:attribute>
+                        <xsl:attribute name="select">
+                            <xsl:value-of select="concat('if (empty($', @name ,')) then null-node{} else $', @name)"/>
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/function-metadata.xsl:100]: @ryanjdew 
This seems to work fine. I didn't have to add xsl:choose element. Let me know if you see any issues with this.

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/mapping-functions/mapping-functions-sequence-parameter/test-data/entities/Customer.entity.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/mapping-functions/mapping-functions-sequence-parameter/test-data/entities/Customer.entity.json:1]: Please use the new createSimpleMappingProject helper in data-hub-test-helper.sjs. You can add the "items" array to the Customer entity model, though let's call it "integers" to make it clear that these are integers. 

The intent of this helper is to cut down on the number of flows/entities/steps that we have to create as separate files.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/function-metadata.xsl (hunk 1 above)
Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/function-metadata.xsl:100]: @srinathgit  Can you submit a separate PR first where this file is included as-is, with a comment indicating the version of ML that it came from? Also include the functions in xquery-lib as they currently are in ML.

We'll get that quickly merged into develop (can do a force merge if needed), and then submit your PR again with the changes to these OOTB stylesheets and functions.

The benefit is that our version history will then make it very clear what changes we made to the OOTB stuff.

**Review by rjrudin — CHANGES_REQUESTED** [review]

**Comment by srinathgit** [comment]
Opened https://github.com/marklogic/marklogic-data-hub/pull/5496, closing this PR

**Pull request closed without merging**